src/piecetype.cpp

```diff
@@ -65,6 +65,21 @@ auto Piece::fromWind(Wind w) -> Piece {
   }
 }
 
+auto Piece::formPiece(Piece::Type suit, uint8_t number) -> Piece {
+  auto valueAppend = number;
+  if (number == 1 || number == 9) {
+    valueAppend |= TERMINAL_BIT;
```

Comment:
You need to check to make sure the bounds of number and return an error if it's outside the correct numbers per suit.

Reply:
Fixed in https://github.com/realliance/libmahjong/commit/0ca2dfbb82dcacdfd84b0b5e54a619783c0dfc7d